src/engine/BMPlayer.php

```diff
@@ -0,0 +1,264 @@
+     * @param int|null $value
+     */
+    protected function set__playerId($value) {
+        if (!is_null($value)) {
+            $this->playerId = intval($value);
+        }
+    }
+
+    /**
+     * Set position
+     *
+     * @param int|null $value
+     */
+    protected function set__position($value) {
+        if (!is_null($value) && !is_int($value)) {
+            throw new InvalidArgumentException('Player position must be an integer or null');
+        }
+
+        if (!is_null($value)) {
+            $this->position = intval($value);
+        }
+    }
+
+    /**
+     * Set button
+     *
+     * @param BMButton $value
+     */
+    protected function set__button($value) {
+        if (!is_null($value) && !($value instanceof BMButton)) {
+            throw new InvalidArgumentException('Button must be a BMButton object');
+        }
+
+        $this->button = $value;
+        if (!is_null($value)) {
+            $this->button->ownerObject = $this->ownerObject;
+            $this->button->playerIdx = $this->position;
+        }
+    }
+
+    /**
+     * Set waitingOnAction
+     *
+     * @param bool $value
+     */
+    protected function set__waitingOnAction($value) {
+        if (!is_bool($value)) {
+            throw new InvalidArgumentException('waitingOnAction must be a boolean');
+        }
+
+        $this->waitingOnAction = $value;
+    }
+
+    /**
+     * Set isPrevRoundWinner
+     *
+     * @param bool $value
+     */
+    protected function set__isPrevRoundWinner($value) {
+        if (!is_bool($value)) {
+            throw new InvalidArgumentException('isPrevRoundWinner must be a boolean');
+        }
+
+        $this->isPrevRoundWinner = $value;
+    }
+
+    /**
+     * Set ownerObject
+     *
+     * @param BMGame $value
+     */
+    protected function set__ownerObject($value) {
+        if (!($value instanceof BMGame)) {
+            throw new InvalidArgumentException('BMPlayer must be owned by a BMGame');
+        }
+
+        $this->ownerObject = $value;
+
+        if ($this->button instanceof BMButton) {
+            $this->button->ownerObject = $value;
+            $this->button->playerIdx = $this->position;
+        }
+    }
+
+    // utility methods
+    /**
+     * Constructor
+     *
+     * @param int    $playerId
+     * @param string $buttonRecipe
+     * @param int    $position
+     */
+    public function __construct(
+        $playerId = NULL,
+        $buttonRecipe = '',
+        $position = NULL
+    ) {
+        $this->set__playerId($playerId);
+        $this->set__position($position);
+        $this->waitingOnAction = FALSE;
+        if (strlen($buttonRecipe) > 0) {
+            $this->set__button(new BMButton);
+            $this->button->load($buttonRecipe);
+        }
+        $this->isPrevRoundWinner = FALSE;
+    }
+
+    /**
+     * Getter
+     *
+     * @param string $property
+     * @return mixed
+     */
+    public function __get($property) {
+        $funcName = 'get__'.$property;
```

Comment:
I just dinged dwvanstone on horizontal whitespace in a different pull, so i feel i should mention it here too.  :>)